code/modules/mapping/space_management/space_reservation.dm

```diff
@@ -70,6 +70,14 @@
 	src.height = height
 	return TRUE
 
+///Change the turf type of all tiles that are belonging to the turf reservation
+/datum/turf_reservation/proc/set_turf_type(new_turf_type)
+	for(var/turf/T AS in reserved_turfs)
+		if(istype(T, turf_type))
```

Comment:
Why this check?

Reply:
Without it, it will change the turf of the ship too. Which is pretty funny, but not the desired effect :)